alacritty_terminal/src/clipboard.rs

```diff
@@ -12,101 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#[cfg(not(any(target_os = "macos", target_os = "windows")))]
-use std::ffi::c_void;
-
-use log::{debug, warn};
-
-use copypasta::nop_clipboard::NopClipboardContext;
-#[cfg(all(not(any(target_os = "macos", windows)), feature = "wayland"))]
-use copypasta::wayland_clipboard;
-#[cfg(all(not(any(target_os = "macos", windows)), feature = "x11"))]
-use copypasta::x11_clipboard::{Primary as X11SelectionClipboard, X11ClipboardContext};
-#[cfg(any(feature = "x11", target_os = "macos", windows))]
-use copypasta::ClipboardContext;
-use copypasta::ClipboardProvider;
-
-pub struct Clipboard {
-    clipboard: Box<dyn ClipboardProvider>,
-    selection: Option<Box<dyn ClipboardProvider>>,
-}
-
-impl Clipboard {
-    #[cfg(any(target_os = "macos", windows))]
-    pub fn new() -> Self {
-        Self::default()
-    }
-
-    #[cfg(not(any(target_os = "macos", windows)))]
-    pub fn new(_display: Option<*mut c_void>) -> Self {
-        #[cfg(feature = "wayland")]
-        {
-            if let Some(display) = _display {
-                let (selection, clipboard) =
-                    unsafe { wayland_clipboard::create_clipboards_from_external(display) };
-                return Self {
-                    clipboard: Box::new(clipboard),
-                    selection: Some(Box::new(selection)),
-                };
-            }
-        }
-
-        #[cfg(feature = "x11")]
-        return Self {
-            clipboard: Box::new(ClipboardContext::new().unwrap()),
-            selection: Some(Box::new(X11ClipboardContext::<X11SelectionClipboard>::new().unwrap())),
-        };
-
-        #[cfg(not(feature = "x11"))]
-        return Self::new_nop();
-    }
-
-    // Use for tests and ref-tests.
-    pub fn new_nop() -> Self {
-        Self { clipboard: Box::new(NopClipboardContext::new().unwrap()), selection: None }
-    }
-}
-
-impl Default for Clipboard {
-    fn default() -> Self {
-        #[cfg(any(feature = "x11", target_os = "macos", windows))]
-        return Self { clipboard: Box::new(ClipboardContext::new().unwrap()), selection: None };
-        #[cfg(not(any(feature = "x11", target_os = "macos", windows)))]
-        return Self::new_nop();
-    }
-}
-
-#[derive(Debug)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
 pub enum ClipboardType {
     Clipboard,
     Selection,
 }
```

Comment:
I'd say this file is a bit short, just for this single enum. Could it maybe be declared and exported in a different place? Probably best to do it in one where it is also used, maybe term/mod.rs?